pkg/types/azure/validation/platform.go

```diff
@@ -106,6 +110,15 @@ func validateAzureStack(p *azure.Platform, fldPath *field.Path) field.ErrorList
 	if p.ARMEndpoint == "" {
 		allErrs = append(allErrs, field.Required(fldPath.Child("armEndpoint"), "ARM endpoint must be set when installing on Azure Stack"))
 	}
+	if p.RHCOSImage == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("RHCOSImage"), "RHCOS Image must be set when installing on Azure Stack"))
+	} else {
+		storageBaseURL := strings.TrimPrefix(p.ARMEndpoint, "https://management.")
+		// If the URL for the image isn't in the ASH environment we can't use it.
+		if !strings.Contains(p.RHCOSImage, storageBaseURL) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("outboundType"), p.RHCOSImage, "RHCOSImage must be in the ASH environment"))
```

Comment:
`outboundType`? Copypasta?

"ASH" -> "Azure Stack". Don't use "ASH" in public-facing messages.